### PR TITLE
Task-27: FE: Aside filter component. Implement brands filter

### PR DESCRIPTION
**Pull request by denys-rybachok**

### CODE REVIEW

File: products.json [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    "categoryId": 1,
+    "subcategoryId": 1,
+    "description": "This is an example product description.",
+    "detailedDescription": [],
+    "price": 225,
+    "quantity": 11,
+    "discount": 5,
+    "seller": 2,
+    "brand": "ZTE",
+    "state": 1,
```
Comment (VYaniv) [inline products.json]: Why don't you store them in the DB?

File: src/constants/api.ts [hunk 2]
```diff
@@ -5,3 +5,6 @@ export const getUserUrl = 'auth/me';
 export const refreshTokensUrl = 'auth/refresh';
 export const logoutUserUrl = 'auth/logout';
 export const getAllProductsUrl = 'product';
+// export const getAllProductsUrl = (minPrice: number, maxPrice: number): string => {
+//     return `product?minPrice=${minPrice}&maxPrice=${maxPrice}`
+// }
```
Comment (VYaniv) [inline src/constants/api.ts]: Why commented?

File: src/hooks/useDebounce.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import {useEffect, useRef, useState} from 'react';
+
+export const useDebounce = (value: any, delay = 500) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  const timerRef: any = useRef();
```
Comment (VYaniv) [inline src/hooks/useDebounce.ts]: `any` -> timer type.

File: src/hooks/useDebounce2.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {useEffect, useState} from 'react';
+
+export const useDebounce2 = (value: any, delay = 500) => {
```
Comment (VYaniv) [inline src/hooks/useDebounce2.ts]: `useDebounce2` ?

File: products.json (hunk 1 above)
Comment (denys-rybachok) [reply products.json]: It should not be here. I am working on it now and removed them. I think I accidently asked to review this PR. I need a few more minutes.

File: src/constants/api.ts (hunk 2 above)
Comment (denys-rybachok) [reply src/constants/api.ts]: Wrong file version from this PR. Now it is changed.

File: products.json (hunk 1 above)
Comment (denys-rybachok) [reply products.json]: Now it is a correct version. 12 files should be. Sorry.

File: src/hooks/useDebounce.ts (hunk 3 above)
Comment (denys-rybachok) [reply src/hooks/useDebounce.ts]: Wrong file version from this PR. Deleted file.

File: src/hooks/useDebounce2.ts (hunk 4 above)
Comment (denys-rybachok) [reply src/hooks/useDebounce2.ts]: Wrong file version from this PR. Deleted file.

**Review by VYaniv — APPROVED** [review]

**Pull request merged**